### PR TITLE
Add "ZW - Sleipnir Mail (anime)"

### PR DESCRIPTION
**Pull request by 42Rundas**
https://yugipedia.com/wiki/ZW_-_Sleipnir_Mail_(anime)

### CODE REVIEW

**Comment by 42Rundas** [comment]
It seems, that I cannot assign the "unofficial" and "new card" labels to my PR's myself.

File: unofficial/c511027107.lua [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+	local ec=c:GetPreviousEquipTarget()
+	return c:IsReason(REASON_LOST_TARGET) and ec and ec:IsReason(REASON_DESTROY)
+end
+
+function s.spfilter(c,e,tp)
+	return c:IsCode(84013237) and c:IsCanBeSpecialSummoned(e,0,tp,false,false)
+end
+
+function s.sptg(e,tp,eg,ep,ev,re,r,rp,chk,chkc)
+	if chkc then return chkc:IsControler(tp) and chkc:IsLocation(LOCATION_GRAVE) and s.spfilter(chkc,e,tp) end
```
Comment (larry126) [inline unofficial/c511027107.lua:126]: You don't need chkc because the effect doesn't target.

File: unofficial/c511027107.lua [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+	if chk==0 then return Duel.GetLocationCount(tp,LOCATION_MZONE)>0
+		and Duel.IsExistingMatchingCard(s.spfilter,tp,LOCATION_GRAVE,0,1,nil,e,tp) end
+	Duel.Hint(HINT_SELECTMSG,tp,HINTMSG_SPSUMMON)
+	local g=Duel.SelectMatchingCard(tp,s.spfilter,tp,LOCATION_GRAVE,0,1,1,nil,e,tp)
+	Duel.SetOperationInfo(0,CATEGORY_SPECIAL_SUMMON,g,1,0,0)
```
Comment (larry126) [inline unofficial/c511027107.lua:131]: As the effect doesn't target, you shouldn't put selecting part here, they should be in the operation.
And in that case, the ```g``` should be a group of all "Number 39: Utopia" in your graveyard.

File: unofficial/c511027107.lua [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+    e1:SetType(EFFECT_TYPE_SINGLE)
+    e1:SetCode(EFFECT_SET_ATTACK_FINAL)
+    e1:SetReset(RESET_EVENT+RESETS_STANDARD)
+    if a:GetControler()==tp then
+        e1:SetValue(d:GetAttack())
+        a:RegisterEffect(e1)
+    else
+        e1:SetValue(a:GetAttack())
+        d:RegisterEffect(e1)
+end
```
Comment (larry126) [inline unofficial/c511027107.lua:110]: line 113 indentation error

File: unofficial/c511027107.lua [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+--equip
+
+function s.filter(c)
+	return c:IsFaceup() and c:IsCode(56840427)
+end
+
+function s.eqtg(e,tp,eg,ep,ev,re,r,rp,chk,chkc)
+	if chkc then return chkc:IsLocation(LOCATION_MZONE) and chkc:IsControler(tp) and s.filter(chkc) end
+	if chk==0 then return Duel.GetLocationCount(tp,LOCATION_SZONE)>0
+		and Duel.IsExistingTarget(s.filter,tp,LOCATION_MZONE,0,1,nil) end
```
Comment (larry126) [inline unofficial/c511027107.lua]: You should carry over the changes from pyrQ's suggestion
Also, you are missing SetOperationInfo in ```eqtg```

File: unofficial/c511027107.lua [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+		and Duel.IsExistingTarget(aux.FilterFaceupFunction(Card.IsCode,56840427),tp,LOCATION_MZONE,0,1,nil) end
+	Duel.Hint(HINT_SELECTMSG,tp,HINTMSG_EQUIP)
+	Duel.SelectTarget(tp,aux.FilterFaceupFunction(Card.IsCode,56840427),tp,LOCATION_MZONE,0,1,1,nil)
+	Duel.SetOperationInfo(0,CATEGORY_EQUIP,e:GetHandler(),1,0,0)
+end
+
+function s.eqop(e,tp,eg,ep,ev,re,r,rp)
+	local c=e:GetHandler()
+	if not c:IsRelateToEffect(e) then return end
+	if c:IsLocation(LOCATION_MZONE) and c:IsFacedown() then return end
```
Comment (larry126) [inline unofficial/c511027107.lua:56]: You should carry over the change I made in Ashura Strike to here

File: unofficial/c511027107.lua [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+function s.atkop(e,tp,eg,ep,ev,re,r,rp)
+    local a=Duel.GetAttacker()
+    local d=Duel.GetAttackTarget()
+    if not a:IsRelateToBattle() or not d:IsRelateToBattle() then return end
+    local e1=Effect.CreateEffect(e:GetHandler())
+    e1:SetOwnerPlayer(tp)
+    e1:SetType(EFFECT_TYPE_SINGLE)
+    e1:SetCode(EFFECT_SET_ATTACK_FINAL)
+    e1:SetReset(RESET_EVENT+RESETS_STANDARD)
+    if a:GetControler()==tp then
```
Comment (larry126) [inline unofficial/c511027107.lua:104]: Use ```IsControler``` instead of ```GetControler```

File: unofficial/c511027107.lua [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+	local ec=e:GetHandler():GetEquipTarget()
+	return ec and (ec==Duel.GetAttacker() or ec==Duel.GetAttackTarget()) and ec:GetBattleTarget() and ec:GetBattleTarget():IsControler(1-tp) and e:GetHandler():GetFlagEffect(id)==0
+end
+
+function s.atkop(e,tp,eg,ep,ev,re,r,rp)
+    local a=Duel.GetAttacker()
+    local d=Duel.GetAttackTarget()
+    if not a:IsRelateToBattle() or not d:IsRelateToBattle() then return end
+    local e1=Effect.CreateEffect(e:GetHandler())
+    e1:SetOwnerPlayer(tp)
```
Comment (larry126) [inline unofficial/c511027107.lua:100]: SetOwnerPlayer isn't required here. This is used in lingering effects that have something to do with player control.

**Pull request merged**